src/client/tcp_dns.js

```diff
@@ -19,19 +19,24 @@ module.exports = function (client, options) {
       // If port was not defined (defaults to 25565), host is not an ip and is not localhost
       if (options.port === 25565 && net.isIP(options.host) === 0 && options.host !== 'localhost') {
         // Try to resolve SRV records for the comain
-        dns.resolveSrv(`_minecraft._tcp.${options.host}`, (err, addresses) => {
-          // Error resolving domain
-          if (err) debug(`srv lookup failed: ${err}`)
+        try {
+          const resolved = await dns.resolveSrv(`_minecraft._tcp.${options.host}`)
+          const [{ name, port }] = resolved
 
-          // SRV Lookup resolved correctly
-          if (addresses && addresses.length > 0) {
-            debug(`srv lookup returned ${addresses}`)
-            options.host = addresses[0].name
-            options.port = addresses[0].port
-          }
-        })
+          debug(`[DNS] SRV Lookup: ${name}:${port}`)
+
+          options.host = name
+          options.port = port
+        } catch (error) {
+          if (error.code && error.code === 'ENOTFOUND') debug('[DNS] Host does not seem to use an SRV record...')
+          else debug(`[DNS] SRV Resolve Failed: ${error}`)
```

Comment:
Does this debug statement work? Is error not an object? You can probably replace the logging with
```suggestion
          catch (error) { debug(error) }
```